geopyspark-backend/geotrellis/src/main/scala/geopyspark/geotrellis/SpatialTiledRasterLayer.scala

```diff
@@ -325,22 +326,29 @@ class SpatialTiledRasterLayer(
   }
 
   def hillshade(
-    sc: SparkContext,
     azimuth: Double,
     altitude: Double,
-    zFactor: Double,
+    zFactorCalculator: ZFactorCalculator,
     band: Int
   ): TiledRasterLayer[SpatialKey] = {
-    val tileLayer = TileLayerRDD(rdd.mapValues(_.band(band)), rdd.metadata)
-
-    implicit val _sc = sc
-
-    val result = tileLayer.hillshade(azimuth, altitude, zFactor)
+    val mt = rdd.metadata.mapTransform
+    val cellSize = rdd.metadata.cellSize
 
-    val multibandRDD: MultibandTileLayerRDD[SpatialKey] =
-      MultibandTileLayerRDD(result.mapValues{ tile => MultibandTile(tile) }, result.metadata)
+    SpatialTiledRasterLayer(
+      zoomLevel,
+      rdd.withContext { rdd =>
+        rdd.bufferTiles(bufferSize = 1).mapPartitions[(SpatialKey, MultibandTile)](
+        { iter =>
+          iter.map { case (key, BufferedTile(tile, bounds)) =>
+            val zfactor = zFactorCalculator.deriveZFactor(mt.keyToExtent(key))
+            val hillshadeTile = Hillshade(tile.bands(band), Square(1), Some(bounds), cellSize, azimuth, altitude, zfactor).interpretAs(ShortConstantNoDataCellType)
```

Comment:
why the call to `interpretAs` here? Is it defensive or otherwise ?

Reply:
That was in the original implementation of the `hillshade` method, so I kept it there. The docs stated that the `hillshade` layer is always has a cell type of `ShortConstantNoDataCellType`, so I assumed it was enforcing that.

Reply:
It makes sense. In current implementation its going to be a NOOP.